src/asset_manager/common.rs

```diff
@@ -0,0 +1,243 @@
+//! Provided common things for asset management
+
+use asset_manager::{Asset, AssetFormat, AssetStore, AssetStoreError};
+
+// -----------------------------------------------------------------------------------------
+// Asset stores
+// -----------------------------------------------------------------------------------------
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::io::{BufReader, Error as IoError, ErrorKind, Read, Seek};
+
+use zip::ZipArchive;
+
+#[cfg(not(android))]
+mod desktop {
+    use super::*;
+
+    use asset_manager::{AssetFormat, AssetStoreError};
+
+    pub fn read_asset<T: Asset, F: AssetFormat>(name: &str,
+                                                format: &F)
+                                                -> Result<Box<[u8]>, AssetStoreError> {
+        read_asset_from_path(Path::new("assets").join(T::category()).join(name), format)
+    }
+
+    pub fn read_asset_from_path<P, F>(path: P, format: &F) -> Result<Box<[u8]>, AssetStoreError>
+        where P: AsRef<Path>,
+              F: AssetFormat
+    {
+        let mut last_error = None;
+
+        let mut path = PathBuf::from(path.as_ref());
+
+        for extension in format.file_extensions() {
+            path.set_extension(extension);
+
+            println!("Trying path {:?}", &path);
```

Comment:
Ah! I was debugging it (because I used `.join()` instead of `set_extension` and forgot to remove it